test/async-wait.c

```diff
@@ -0,0 +1,123 @@
+/********************************************************************
+
+   Copyright 2013 Konstantin Olkhovskiy <lupus@oxnull.net>
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+ ********************************************************************/
+
+#include <ev.h>
+#include <check.h>
+#include <evfibers_private/fiber.h>
+
+#include "async-wait.h"
+
+struct fiber_arg {
+	ev_async *async;
+	fbr_id_t *fibers;
+};
+
+static int async_recvd = 0;
+
+static void async_sender(FBR_P_ void *_arg)
+{
+	struct fiber_arg *arg = _arg;
+	ev_async *async = arg->async;
+
+	/* wait briefly to make sure other fiber blocks */
+	fbr_sleep(fctx, 1);
+
+	/* send async event in libev to wake up other fiber */
+	ev_async_send(EV_DEFAULT, async);
+	return;
+}
+
+static void async_waiter(FBR_P_ void *_arg)
+{
+	struct fiber_arg *arg = _arg;
+	ev_async *async = arg->async;
+
+	fbr_async_wait(fctx, async);
+
+	async_recvd = 1;
+
+	return;
+}
+
+static void async_handler_cb(EV_P_ ev_async *w, int revents)
+{
+	/* What should be done when async event received. */
+
+	/* look at args just to hush warnings */
+	fail_if(NULL == w);
+	fail_if(EV_ERROR == revents);
```

Comment:
Travis tests complain about this line. You could possible use void cast warning suppression, or change the test here to a flag test (as revents is a bitmask):

```c
    fail_if(EV_ERROR & revents);
```